wolfcrypt/src/asn.c

```diff
@@ -13956,8 +13979,7 @@ static int DecodeAltNames(const byte* input, int sz, DecodedCert* cert)
             XMEMCPY(dnsEntry->name, &input[idx], strLen);
             dnsEntry->name[strLen] = '\0';
 
-            dnsEntry->next = cert->altNames;
-            cert->altNames = dnsEntry;
+            AddAltName(cert, dnsEntry);
```

Comment:
Why is this one missing `dnsEntry->next = NULL;`? I don't see a memset or init for it.... The uriEntry and ipAddr do this.

Reply:
Those should be deleted. The functionality was moved into the subroutine.

```
static void AddAltName(DecodedCert* cert, DNS_entry* dnsEntry)
{
#if defined(OPENSSL_EXTRA) && !defined(WOLFSSL_ALT_NAMES_NO_REV)
    dnsEntry->next = NULL;
```